table/tables/index.go

```diff
@@ -204,26 +204,48 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 	if err != nil {
 		return 0, err
 	}
+
+	ctx := opt.Ctx
+	if opt.Untouched {
+		// If the index kv was untouched(unchanged), and the key/value already exists in mem-buffer,
+		// no need to re-write the index kv.
+		txn, err1 := sctx.Txn(true)
+		if err1 != nil {
+			return 0, err1
+		}
+		_, err = txn.GetMemBuffer().Get(ctx, key)
```

Comment:
Get the key in membuffer to confirm it exists? If the err is not nil, it will be overwritten at line 230.

Reply:
Yes, should make sure the key was not in mem-buffer to avoid overwriting it.
The error could only be `ErrNotExists`.

Reply:
I will update the comment here.

Reply:
ok